lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/core/nodes/TokenizedPhraseQueryNode.java

```
@@ -70,10 +70,8 @@ public QueryNode cloneTree() throws CloneNotSupportedException {
   @Override
   public CharSequence getField() {
     List<QueryNode> children = getChildren();
-
-    if (children == null || children.size() == 0) {
+    if (children == null || children.isEmpty()) {
       return null;
-
     } else {
       return ((FieldableNode) children.get(0)).getField();
```

According to the `setField(CharSequence)` method below children are not necessarily `FieldableNode`.
I.e. this line might throw `ClassCastException`?

---

I see your point. Maybe it's worth adding an assertion?

---

I think the two methods should be aligned.
If one has an assert (assumes all the children are `FieldableNode`) then the other should do so too.
Of they both should silently ignore children of other types.

I would probably go with "silently ignore children of other types" as the least surprising option.

---

I think having the two methods aligned, the way they are in your latest commit, makes sense.